lib/logstash/inputs/elasticsearch.rb

```diff
@@ -160,38 +157,42 @@ def run(output_queue)
       r = process_next_scroll(output_queue, r['_scroll_id'])
       has_hits = r['has_hits']
     end
-  end # def run
+  end
 
   private
+
   def process_next_scroll(output_queue, scroll_id)
     r = scroll_request(scroll_id)
     r['hits']['hits'].each { |hit| push_hit(hit, output_queue) }
     {'has_hits' => r['hits']['hits'].any?, '_scroll_id' => r['_scroll_id']}
   end
 
-  private
   def push_hit(hit, output_queue)
     event = LogStash::Event.new(hit['_source'])
     decorate(event)
 
     if @docinfo
-      event[@docinfo_target] ||= {}
+      # do not assume event[@docinfo_target] to be in-place updatable. first get it, update it, then at the end set it in the event.
+      docinfo_target = event[@docinfo_target] || {}
 
-      unless event[@docinfo_target].is_a?(Hash)
-        @logger.error("Elasticsearch Input: Incompatible Event, incompatible type for the `@metadata` field in the `_source` document, expected a hash got:", :metadata_type => event[@docinfo_target].class)
+      unless docinfo_target.is_a?(Hash)
```

Comment:
well, if its not a hash, it is because it is an existing value and the only thing we can do is:
- overwrite and log
- not overwrite/ignore and log
- raise and log

overwriting will destroy an existing value and raising crashes the whole thing. since it currently raises, changing that will change the behaviour of the plugin, but on the other hand, I think this really is an edge case. 

so, I'm on the fense. I'd rather stay on the "safe" side and keep curent behaviour and re-evaluate later in another issue?